src/scripts/main.js

```diff
@@ -1,3 +1,317 @@
+        type="number"
+        data-qa="salary"
+      />
+    </label>
+    <button
+      type="button"
+      id="save-button"
+    >
+      Save to table
+    </button>
+  </form>
+  `;
+
+table.insertAdjacentHTML('afterend', formHtml);
+
+// 4. Show notification if form data is invalid
+function pushNotification(message, type) {
+  const notification = document.createElement('div');
+  const titleSpan = document.createElement('span');
+
+  notification.className = `notification ${type}`;
+  notification.setAttribute('data-qa', 'notification');
+
+  titleSpan.className = 'title';
+  titleSpan.textContent = type.toUpperCase();
+
+  notification.appendChild(titleSpan);
+  notification.innerHTML += ` <span>${message}</span>`;
+  document.body.appendChild(notification);
+
+  setTimeout(() => {
+    notification.style.display = 'none';
+  }, 2000);
+}
+
+function validateName(formName) {
+  const nameRegex = /^[A-Za-z\s]+$/;
+
+  if (formName.length < 4 || !nameRegex.test(formName)) {
+    pushNotification(
+      'Name must be at least 4 characters long and contain only letters.',
+      'error',
+    );
+
+    return false;
+  }
+
+  return true;
+}
+
+function validateAge(age) {
+  if (age < 18 || age > 90) {
+    pushNotification('Age must be between 18 and 90.', 'error');
+
+    return false;
+  }
+
+  return true;
+}
+
+function validatePosition(position) {
+  if (!position.trim()) {
+    pushNotification('Position is required.', 'error');
+
+    return false;
+  }
+
+  return true;
+}
+
+function validateOffice(office) {
+  if (!office.trim()) {
+    pushNotification('Office is required.', 'error');
+
+    return false;
+  }
+
+  return true;
+}
+
+function validateSalary(salary) {
+  const parsedSalary = parseFloat(salary);
+
+  if (isNaN(parsedSalary) || parsedSalary <= 0) {
+    pushNotification('Salary must be a valid positive number.', 'error');
+
+    return false;
+  }
+
+  return true;
+}
+
+function validateFormData(formData) {
+  return (
+    validateName(formData.name) &&
+    validateAge(formData.age) &&
+    validatePosition(formData.position) &&
+    validateOffice(formData.office) &&
+    validateSalary(formData.salary)
+  );
+}
+
+const form = document.querySelector('.new-employee-form');
+
+document.getElementById('save-button').addEventListener('click', () => {
+  const formData = [...form.elements].reduce((acc, formInput) => {
+    if (formInput.name) {
+      acc[formInput.name] = formInput.value;
+    }
+
+    return acc;
+  }, {});
+
+  if (validateFormData(formData)) {
+    const newRow = document.createElement('tr');
+
+    newRow.innerHTML = `
+      <td>${formData.name}</td>
+      <td>${formData.position}</td>
+      <td>${formData.office}</td>
+      <td>${formData.age}</td>
+      <td>$${parseInt(formData.salary).toLocaleString('en-US')}</td>`;
+
+    table.querySelector('tbody').appendChild(newRow);
+    pushNotification('Employee added successfully!', 'success');
+    form.reset();
+  }
+});
```

Comment:
Code quality issue: This event listener is quite large and doing a lot of things, consider breaking it down into smaller functions for better readability and maintainability.